internal/service/advancedcluster/resource_advanced_cluster.go

```diff
@@ -850,6 +850,10 @@ func expandAdvancedReplicationSpec(tfMap map[string]any) *matlas.AdvancedReplica
 		RegionConfigs: expandRegionConfigs(tfMap["region_configs"].([]any)),
 	}
 
+	if tfMap["id"].(string) != "" {
```

Comment:
Just to confirm, in we are setting the id in update request of `mongodbatlas_cluster` resource right?

Reply:
@AgustinBettati this method is called in both update and create. The reason why it should only run on the update is because of the if itself. I am assuming that at the creation phase this field is empty. Any concerns?

Reply:
The function is called `expandAdvancedReplicationSpec` but this PR is also updating the cluster test 🤔 . Is `expandAdvancedReplicationSpec` called also in the cluster resource? if not, is the cluster resource already sending the `id` during the update?

Reply:
Yes that statement seems correct to me. My doubt was more related to how we handle this id in `mongodbatlas_cluster`, not a blocker just something to keep in mind.

Reply:
> The function is called expandAdvancedReplicationSpec but this PR is also updating the cluster test
@andreaangiolillo @AgustinBettati 

Adding tests to `cluster` was more like to increase test coverage. Sorry if it created confusion. confirming that `expandAdvancedReplicationSpec` is only affecting `advanced_cluster`.

regarding cluster: that is a completely different story: while in here we use list, in that one we are representing replication_specs as a set and we're also using v1.0 - I added the test on purpose: to see if it was able to handle replicationspecs updates and indeed it does without having to change the logic.